osfbot/exts/moderation/logs.py

```diff
@@ -0,0 +1,243 @@
+from datetime import datetime
+from typing import List, Optional
+
+from disnake import (
+    ButtonStyle,
+    Component,
+    Embed,
+    Member,
+    Message,
+    RawMessageDeleteEvent,
+    TextChannel,
+)
+from disnake.ext.commands import Cog
+from disnake.ui import Button
+from disnake.utils import format_dt
+from loguru import logger
+
+from ...bot import Bot
+from ...constants import Channels, Colors
+
+
+class ModerationLogs(Cog):
+    """Used to log infractions and important actions in the specified channel."""
+
+    def __init__(self, bot: Bot):
+        self.bot = bot
+        self.log_channel: Optional[TextChannel] = None
+        super().__init__()
+
+    async def post_message(
+        self,
+        embed: Embed,
+        components: List[Component] = None,
+    ) -> Optional[Message]:
+        """Send the given message in the log channel."""
+        if not self.log_channel:
+            await self.bot.wait_until_ready()
+            self.log_channel = await self.bot.fetch_channel(Channels.log)
+
+            if not self.log_channel:
+                logger.error(f"Failed to get log channel with ID ({Channels.log})")
+
+        return await self.log_channel.send(embed=embed, components=components)
+
+    async def post_formatted_message(
+        self,
+        actor: Member,
+        title: str,
+        color: int = Colors.green,
+        fields: Optional[list] = None,
+        url: Optional[str] = None,
+        components: List[Component] = None,
+    ) -> None:
+        """Formats an embed to be posted in the log channel."""
+        embed = Embed(
+            title=title,
+            color=color,
+            timestamp=datetime.now(),
+        ).set_thumbnail(url=actor.display_avatar.url)
+
+        if url:
+            embed.url = url
+
+        for field in fields:
+            embed.add_field(
+                name=field["name"], value=field["value"], inline=field["inline"]
+            )
+
+        await self.post_message(embed=embed, components=components)
+
+    @Cog.listener()
+    async def on_raw_message_delete(self, payload: RawMessageDeleteEvent) -> None:
```

Comment:
It would be useful to not delete a command deleted by the bot.
For example : `>dm <user> <message>` is deleted by the bot and logged too.

Reply:
As in, not have the command deleted at all?
I can disable logging for commands, but deleting the `>dm` command and showing relevant info in the embed makes for a cleaner experience.

Reply:
Oh, I meant to say - not log a command deleted by the bot.
So yes, disable logging for commands.

Reply:
This has been fixed by converting is to a slash command, as discussed.